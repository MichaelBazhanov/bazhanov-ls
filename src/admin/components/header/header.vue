<template>
	<div class="header-component">

		<headline title="Панель администрирования" @logout='logout'>
			<user></user>
		</headline>

		<navigation/>
	</div>
</template>

<script>
import user from "../../components/user/user"; //импорт компонента
import headline from "../../components/headline/headline"; //импорт компонента
import navigation from "../../components/navigation/navigation"; //импорт компонента

import { mapActions } from 'vuex'

export default {
	components: {
		headline,
		user,
		navigation,
	},
	methods: {
		...mapActions({
			logout: 'user/logout'
		}),
	}
}
</script>

<style lang="postcss" scoped src="./header.pcss">
</style>>
