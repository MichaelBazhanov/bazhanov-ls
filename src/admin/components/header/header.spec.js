import header from './header'
import { mount } from "@vue/test-utils"

it('header snapshot', async () => {
	//steps
	const wrapper = mount(header, { stubs: ['router-link', 'router-view'] })

	//actions
	await wrapper.vm.$emit('logout', 'logout')

	//check
	await wrapper.vm.$nextTick(); //ожидаем асинхронного обновления всего
	expect(wrapper.emitted('logout')[0]).toEqual(['logout'])
	expect(wrapper.element).toMatchSnapshot()
})