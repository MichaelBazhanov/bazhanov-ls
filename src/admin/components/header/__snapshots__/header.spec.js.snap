// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header snapshot 1`] = `
<div
  class="header-component"
>
  <div
    class="headline-component"
  >
    <div
      class="container container-headline"
    >
      <div
        class="user-components"
      >
        <div
          class="avatar-component"
          src="https://picsum.photos/300/300"
          style="width: 2.7rem; height: 2.7rem; min-width: 2.7rem; min-height: 2.7rem;"
        >
          <img
            alt="user picture"
            class="image"
            src="https://picsum.photos/300/300"
          />
        </div>
         
        <div
          class="content"
        >
          <div
            class="username"
          >
            Бажанов Михаил
          </div>
           
          <div
            class="btns"
          >
            <button
              class="btn"
              type="button"
            >
              Выйти
            </button>
          </div>
        </div>
      </div>
       
      <div
        class="title"
      >
        Панель администрирования
      </div>
       
      <div
        class="btns"
      >
        <button
          class="btn"
          type="button"
        >
          Выйти
        </button>
      </div>
    </div>
  </div>
   
  <div
    class="navigation-component"
  >
    <div
      class="container"
    >
      <ul
        class="list"
      >
        <li
          class="item active"
        >
          <router-link-stub
            class="link"
            tag="a"
            to="/about"
          >
            
					Обо мне
				
          </router-link-stub>
        </li>
        <li
          class="item"
        >
          <router-link-stub
            class="link"
            tag="a"
            to="/values"
          >
            
					Навыки
				
          </router-link-stub>
        </li>
        <li
          class="item"
        >
          <router-link-stub
            class="link"
            tag="a"
            to="/works"
          >
            
					Работы
				
          </router-link-stub>
        </li>
        <li
          class="item"
        >
          <router-link-stub
            class="link"
            tag="a"
            to="/reviews"
          >
            
					Отзывы
				
          </router-link-stub>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
